### PR TITLE
Uuv bluerov2 extension

### PR DESCRIPTION
**Pull request by Zarbokk**
**Describe problem solved by this pull request**
The current uuv_attitude_control module does not support the 6-DOF BlueROV2 on a fundamental level

**Describe your solution**
Added a new control Module for the BlueROV2 which stabilizes the robot(RPY and DEPTH) with a very simple controller.
Additionally, it is possible to move the robot on the x- and y-axis.

The Mixer files are changed and in the SITL project, an additional pull request is created, which changes the SDF files from the hippocampus and the BlueROV2
 

**Describe possible alternatives**
None

**Test data / coverage**
It was tested only in the Gazebo Environment. Due to the Covid situation, the testing on the Robot follows in the following weeks/months.

### CODE REVIEW

**Comment by Jaeyoung-Lim** [comment]
@lennartalff Would you be able to review this PR?

File: boards/px4/fmu-v4/ctrlalloc.cmake [hunk 1]
```diff
@@ -91,6 +91,7 @@ px4_add_board(
 		sih
 		temperature_compensation
 		uuv_att_control
+		uuv_bluerov2_control
```
Comment (Jaeyoung-Lim) [inline boards/px4/fmu-v4/ctrlalloc.cmake:94]: Why is a controller just for bluerov2 required to be a separate module?

Is there a reason that you couldn't integrate this into the `uuv_att_control` module?

**Review by Jaeyoung-Lim — COMMENTED** [review]
It would be better if we could integrate this into the `uuv_att_control` module, so that we can make it more generic than just the blue rov.

Would it be possible to integrate the RPY control into the `uuv_att_control` and maybe add the depth control logic to a separate module e.g. `uuv_pos_control` module?

@lennartalff @DanielDuecker Would you also be able to review this PR?

**Comment by DanielDuecker** [comment]
Hi, 
Thanks for reaching out!
Great that people keep pushing PX4 to the underwater domain!

Let's sync and push this together! 
We (@lennartalff , @nbauschmann, myself, ...) recently invested quite some work on the PX4<->BlueRov integration to use PX4 in a uuv-robotics-class at our school (maybe you find this interesting: https://hippocampusrobotics.github.io/fav_docs/index.html. However, focus was -so far- more on the ROS-side and  resulting in some class specific tweeks due to time limitations) 

Anyhow, it would be great to identify synergies between the efforts on zarbokk's and our side!

I am not sure whether we want to add modules which are specific to single platform such as BlueRov2. Hope we find a more general solution, e.g. via a pos_module where this is possible.

I will reach out to @Zarbokk and @lennartalff so we can discuss this and keep you and the others in the loop @Jaeyoung-Lim

**Comment by DanielDuecker** [comment]
Short update, we (@Zarbokk @lennartalff + myself) came to the conclusion to go for a general uuv_pos_control module. @Zarbokk will put his great extensions to this new module. Addtionally, this approach requires slight extensions to the existing uuv_att_control module. However, this aligns more with the modular design of other vehicles.

**Comment by Jaeyoung-Lim** [comment]
Closing in favor of https://github.com/PX4/PX4-Autopilot/pull/16688

**Pull request closed without merging**